apps/flutter_exercise/lib/views/widgets/login_widgets/form.dart

```diff
@@ -0,0 +1,231 @@
+      style: TextStyle(
+          color: Color.fromARGB(255, 255, 0, 0), fontWeight: FontWeight.bold),
+    )
+  ]);
+}
+
+// Funcion que devuelve widget campo de formulario
+Widget _textForm(String tipo, String pLabelText) {
+  return Padding(
+    padding: const EdgeInsets.all(8.0),
+    child: Material(
+      elevation: 5.0,
+      borderRadius: BorderRadius.circular(10.0),
+      shadowColor: const Color.fromARGB(255, 91, 118, 92),
+      child: TextFormField(
+        decoration: InputDecoration(
+            fillColor: const Color.fromARGB(255, 247, 255, 247),
+            filled: true,
+            contentPadding: const EdgeInsets.all(10.0),
+            prefixIcon: _getIcon(tipo),
+            labelText: "$pLabelText $tipo",
+            border: OutlineInputBorder(
+                borderRadius: BorderRadius.circular(10.0),
+                borderSide: BorderSide.none)),
+        // The validator receives the text that the user has entered.
+        validator: (value) {
+          if (value == null || value.isEmpty) {
+            return '$pLabelText $tipo';
+          }
+          return null;
+        },
+      ),
+    ),
+  );
+}
+
+// Función para obtener prefixIcon de _textForm
+Icon _getIcon(String tipo) {
+  switch (tipo) {
+    case 'nombre':
+      return const Icon(Icons.account_circle_outlined,
+          color: Color.fromARGB(255, 91, 118, 92));
+    case 'email':
+      return const Icon(Icons.email_outlined,
+          color: Color.fromARGB(255, 91, 118, 92));
+    case 'contraseña':
+      return const Icon(Icons.key, color: Color.fromARGB(255, 91, 118, 92));
+    default:
+      return const Icon(null);
+  }
+}
+
+// Funcion que devuelve Column con _textLabel y _textForm
+Widget _fieldForm(String tipo, [String pLabelText = "Introduce tu "]) {
+  return Column(
+    children: [_textLabel(tipo), _textForm(tipo, pLabelText)],
+  );
+}
+
+// Funcion que devuelve botón de confirmar
+Widget _comenzarButton(GlobalKey<FormState> key, BuildContext context) {
+  return Padding(
+    padding: const EdgeInsets.symmetric(vertical: 16),
+    child: ElevatedButton(
+      onPressed: () {
+        // Validate returns true if the form is valid, or false otherwise.
+        if (key.currentState!.validate()) {
+          // If the form is valid, display a snackbar. In the real world,
+          // you'd often call a server or save the information in a database.
+          ScaffoldMessenger.of(context).showSnackBar(
+            const SnackBar(content: Text('Processing Data')),
+          );
+        }
+      },
+      style: ElevatedButton.styleFrom(
+          backgroundColor: const Color.fromARGB(255, 113, 144, 125),
+          padding: const EdgeInsets.symmetric(
+              horizontal: 60, vertical: 10)), // set the background color
+      child: const Text('Comenzar',
+          style:
+              TextStyle(color: Color.fromARGB(255, 23, 57, 25), fontSize: 20)),
+    ),
+  );
+}
+
+enum Formulario { name, email, pass }
```

Comment:
@Mazit0 if you create an enum, that is a good idea, you should use it.

Reply:
lo habia puesto para que al usar el método _fieldForm lo tuviese de parametro y asi usar un switch para pasar los parametros a los metodos que carga _fieldForm pero al final se me olvidó xD